File: src/main/java/com/skylotoneoneone/service/PermissionService.java
package com.skylotoneoneone.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skylotoneoneone.domain.Permission;
import com.skylotoneoneone.dto.PermissionDTO;
import com.skylotoneoneone.dto.PermissionSearchDTO;
import com.skylotoneoneone.dto.PermissionPageDTO;
import com.skylotoneoneone.dto.PermissionConvertCriteriaDTO;
import com.skylotoneoneone.service.GenericService;
import com.skylotoneoneone.dto.common.RequestDTO;
import com.skylotoneoneone.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface PermissionService extends GenericService<Permission, Integer> {

	List<Permission> findAll();

	ResultDTO addPermission(PermissionDTO permissionDTO, RequestDTO requestDTO);

	ResultDTO updatePermission(PermissionDTO permissionDTO, RequestDTO requestDTO);

    Page<Permission> getAllPermissions(Pageable pageable);

    Page<Permission> getAllPermissions(Specification<Permission> spec, Pageable pageable);

	ResponseEntity<PermissionPageDTO> getPermissions(PermissionSearchDTO permissionSearchDTO);
	
	List<PermissionDTO> convertPermissionsToPermissionDTOs(List<Permission> permissions, PermissionConvertCriteriaDTO convertCriteria);

	PermissionDTO getPermissionDTOById(Integer permissionId);







}

File: skylotoneoneone_ui/src/services/PermissionService.js
import http from "../http-common"; 

class PermissionService {
  getAllPermissions(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/permission/permissions`, searchDTO);
  }

  get(permissionId) {
    return this.getRequest(`/permission/${permissionId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/permission?field=${matchData}`, null);
  }

  addPermission(data) {
    return http.post("/permission/addPermission", data);
  }

  update(data) {
  	return http.post("/permission/updatePermission", data);
  }
  
  uploadImage(data,permissionId) {
  	return http.postForm("/permission/uploadImage/"+permissionId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new PermissionService();
